fish: replace use of tr with string split

bouk opened the pull request:
This shaves about 4ms off fish start time. It also makes it more clear what's happening.

Before (profiled with 'fish --profile prof.txt -c fish_prompt'):

```
225	4636	----> set -l __nix_profile_paths (echo $NIX_PROFILES | /nix/store/m5ajgnzp2512na31brwfmydwk3l1gawb-coreutils-8.31/bin/tr ' ' '\n')[-1..1]
4411	4411	-----> echo $NIX_PROFILES | /nix/store/m5ajgnzp2512na31brwfmydwk3l1gawb-coreutils-8.31/bin/tr ' ' '\n'
```

After:

```
190	248	----> set -l __nix_profile_paths (string split ' ' $NIX_PROFILES)[-1..1]
```

Also use -p when expanding the profile paths, to make it clear we are
prepending.


###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cole-h reviewed (approved):
Diff LGTM, builds and runs fine.

However, I'd like to ask that you split your changes into two commits, `fish: replace tr with builtin string split` and `fish: prepend paths in fishPreInitHooks`, since they are two unrelated changes (I'm bad at doing this myself).

```
[3 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/90670
1 package built:
fish
```

bouk committed:
fish: replace use of tr with string split

This shaves about 4ms off fish start time.

Before (profiled with 'fish --profile prof.txt -c fish_prompt'):

225	4636	----> set -l __nix_profile_paths (echo $NIX_PROFILES | /nix/store/m5ajgnzp2512na31brwfmydwk3l1gawb-coreutils-8.31/bin/tr ' ' '\n')[-1..1]
4411	4411	-----> echo $NIX_PROFILES | /nix/store/m5ajgnzp2512na31brwfmydwk3l1gawb-coreutils-8.31/bin/tr ' ' '\n'

After:

190	248	----> set -l __nix_profile_paths (string split ' ' $NIX_PROFILES)[-1..1]

bouk committed:
fish: use -p to prepend profile directories

bouk commented:
You're right, I've split the commit in two

rnhmjoj reviewed (approved):
Looks good and fish works as expected.
